src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,50 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private int range = 40;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private final String[] figures = {"square",
+            "rectangle",
+            "righttriangle",
+            "circle",
+            "isoscelestrapezoid"};
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int randomFigureIndex = random.nextInt(figures.length);
+        String figure = figures[randomFigureIndex];
+
+        if (figure.equals("square")) {
+            Figure square = new Square(random.nextInt(range), colorSupplier.getRandomColor());
+            return square;
+        } else if (figure.equals("rectangle")) {
+            Figure rectangle = new Rectangle(random.nextInt(range),
+                    random.nextInt(range),
+                    colorSupplier.getRandomColor());
```

Comment:
Creating a figure, don't pass expressions in the constructor. Create separate variables for the dimensions and the color, and then pass them to the constructor.